app/Http/Controllers/CarController.php

```diff
@@ -101,10 +110,33 @@ public function update(Request $request, string $id)
             'carTitle' => 'required|string',
             'description' => 'required|string|max:1000',
             'price' => 'required|numeric',
+            'image' => 'nullable|image|mimes:jpeg,png,jpg,gif|max:2048',
         ]);
-        $data['published'] = isset($request->published);
+
+        $car = Car::findOrFail($id);
+
+        if ($request->hasFile('image')) {
+
+            $oldImagePath = public_path('assets/images/' . $car->image);
+
+            if (File::exists($oldImagePath)) {
+                File::delete($oldImagePath);
+            }
+
+            $file_extension = $request->image->getClientOriginalExtension();
+            $file_name = time() . '.' . $file_extension;
+            $path = 'assets/images';
+            $request->image->move($path, $file_name);
+            $data['image'] = $file_name;
+
+        } else {
```

Comment:
actually,, no need for else block

if you didn't send one column to update() method, it won't change